components/writers/src/main/java/org/datacleaner/beans/writers/UpdateDataHubAnalyzer.java

```diff
@@ -0,0 +1,570 @@
+import org.datacleaner.api.Validate;
+import org.datacleaner.components.categories.WriteSuperCategory;
+import org.datacleaner.components.convert.ConvertToBooleanTransformer;
+import org.datacleaner.components.convert.ConvertToNumberTransformer;
+import org.datacleaner.connection.CsvDatastore;
+import org.datacleaner.connection.FileDatastore;
+import org.datacleaner.connection.SchemaNavigator;
+import org.datacleaner.connection.UpdateableDatastore;
+import org.datacleaner.connection.UpdateableDatastoreConnection;
+import org.datacleaner.data.MetaModelInputColumn;
+import org.datacleaner.descriptors.FilterDescriptor;
+import org.datacleaner.descriptors.TransformerDescriptor;
+import org.datacleaner.desktop.api.PrecedingComponentConsumer;
+import org.datacleaner.job.builder.AnalysisJobBuilder;
+import org.datacleaner.util.WriteBuffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Named("Update DataHub")
+@Description("Update Golden Records in the DataHub. This component allows you to update fields defined in the golden records metadata. The component always updates a golden record based"
+        + "\non its golden record id."
+        + "\nThe 'values to update' property lets you specify which values must be changed. This component can not update fields of the metadata, like traffic lights or severities or track codes.")
+@Categorized(superCategory = WriteSuperCategory.class)
+@Concurrent(true)
+public class UpdateDataHubAnalyzer implements Analyzer<WriteDataResult>, Action<Iterable<Object[]>>, HasLabelAdvice,
+        PrecedingComponentConsumer {
+
+    private static final String PROPERTY_NAME_VALUES = "Values";
+    private static final String PROPERTY_NAME_CONDITION_VALUES = "Condition values";
+
+    private static final File TEMP_DIR = FileHelper.getTempDir();
+
+    private static final String ERROR_MESSAGE_COLUMN_NAME = "update_table_error_message";
+
+    private static final Logger logger = LoggerFactory.getLogger(UpdateDataHubAnalyzer.class);
+
+    @Inject
+    @Configured(value = PROPERTY_NAME_VALUES, order = 1)
+    @Description("Values to update in the table")
+    InputColumn<?>[] values;
+
+    @Inject
+    @Configured(order = 2)
+    @Description("Names of columns in the target table, on which the values will be updated.")
+    @ColumnProperty
+    @MappedProperty(PROPERTY_NAME_VALUES)
+    String[] columnNames;
+
+    @Inject
+    @Configured(value = PROPERTY_NAME_CONDITION_VALUES, order = 3)
+    @Description("Values that make up the condition of the table update")
+    InputColumn<?>[] conditionValues;
+
+    @Inject
+    @Configured(order = 4)
+    @Description("Names of columns in the target table, which form the conditions of the update.")
+    @ColumnProperty
+    @MappedProperty(PROPERTY_NAME_CONDITION_VALUES)
+    String[] conditionColumnNames;
+
+    @Inject
+    @Configured(order = 5)
+    @Description("Datastore to write to")
+    UpdateableDatastore datastore;
```

Comment:
Looking at the type here, it seems to be generically working for any updateable datastore. That's fine, but I would prefer not having two analyzers that does the same thing.